pkg/operator/render.go

```diff
@@ -51,51 +53,33 @@ func toYAML(i interface{}) []byte {
 	return out
 }
 
-type installConfigGetter func() (installertypes.InstallConfig, error)
-
-func discoverMCOConfig(f installConfigGetter) (*mcfgv1.MCOConfig, error) {
-	ic, err := f()
-	if err != nil {
-		return nil, err
+func createDiscoveredControllerConfigSpec(infra *configv1.Infrastructure, network *configv1.Network) (*mcfgv1.ControllerConfigSpec, error) {
+	if len(network.Spec.ServiceNetwork) == 0 {
+		return nil, fmt.Errorf("service cidr is empty in Network")
 	}
-	dnsIP, err := clusterDNSIP(ic.Networking.ServiceCIDR.String())
+	dnsIP, err := clusterDNSIP(network.Spec.ServiceNetwork[0])
 	if err != nil {
 		return nil, err
 	}
 
-	platform, err := platformFromInstallConfig(ic)
-	if err != nil {
-		glog.Warningf("Warning: %v, using %s", err, platform)
+	platform := "none"
+	switch infra.Status.Platform {
+	case configv1.AWSPlatform:
+		platform = "aws"
+	case configv1.OpenStackPlatform:
+		platform = "openstack"
+	case configv1.LibvirtPlatform:
+		platform = "libvirt"
 	}
 
-	return &mcfgv1.MCOConfig{
-		Spec: mcfgv1.MCOConfigSpec{
-			ClusterDNSIP:        dnsIP,
-			CloudProviderConfig: "",
-			ClusterName:         ic.ObjectMeta.Name,
-			Platform:            platform,
-			BaseDomain:          ic.BaseDomain,
-			SSHKey:              ic.SSHKey,
-		},
+	return &mcfgv1.ControllerConfigSpec{
+		ClusterDNSIP:        dnsIP,
+		CloudProviderConfig: "",
+		EtcdDiscoveryDomain: infra.Status.EtcdDiscoveryDomain,
+		Platform:            platform,
 	}, nil
 }
 
-func platformFromInstallConfig(ic installertypes.InstallConfig) (string, error) {
-	// TODO: these constants are wrong, they should match what is reported by the infrastructure provider
```

Comment:
Do we still need this TODO? If so let's move it on up to line 64

Reply:
Nope.